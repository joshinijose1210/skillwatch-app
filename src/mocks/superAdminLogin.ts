export const SuperAdminLoginMockData = {
    get: {
        url: 'https://dummy-api.example.com/api/employees/email?emailId=superadmin@company.com',
        responseData: {
            id: 13,
            emailId: 'superadmin@company.com',
            firstName: 'First',
            lastName: 'Last',
            modulePermission: [
                {
                    moduleId: 1,
                    moduleName: 'Organisations',
                    view: true,
                    edit: true
                }
            ]
        }
    },
    post: {
        url: 'https://dummy-api.example.com/login/super-admin',
        responseData: {
            authentication: {
                name: 'superadmin@company.com',
                attributes: {
                    access_token: 'test_access_token',
                    refresh_token: 'test_refresh_token',
                    login_flow: 'Username,Password',
                    token_type: 'Bearer',
                    expires_in: 86400,
                    roles: ['SUPER_ADMIN']
                }
            },
            authenticated: true
        }
    },
    initialState: { isLoggedIn: false }
};
