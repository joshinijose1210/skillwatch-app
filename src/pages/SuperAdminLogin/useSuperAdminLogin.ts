import { ErrorType } from '@common';
import { routeConstants } from '@constants';
import { addToast } from '@medly-components/core';
import { useAppDispatch, useAppSelector } from '@slice';
import { useGetAccessTokenMutation, useGetEmployeeQuery } from '@slice/services';
import { addUser } from '@slice/user';
import { performValidation } from '@utils/validations';
import { useCallback, useEffect, useMemo, useState } from 'react';
import { useLocation, useNavigate } from 'react-router-dom';

export const useSuperAdminLogin = () => {
    const [getAccessToken, { isLoading: isFetchingToken, data: tokenData, error: loginError }] = useGetAccessTokenMutation();

    const { state }: any = useLocation(),
        userDetails = useAppSelector(state => state.user),
        dispatch = useAppDispatch(),
        navigateTo = useNavigate();

    const [email, setEmail] = useState(''),
        [clicked, setClicked] = useState(false),
        [passwordVisible, setPasswordVisible] = useState(false);

    const [userData, setUserData] = useState({
        password: '',
        email: '',
        errors: {
            password: '',
            email: ''
        }
    });

    const {
        data,
        isLoading: isUserDataLoading,
        error
    } = useGetEmployeeQuery(
        {
            path: 'email',
            params: [{ name: 'emailId', value: email }]
        },
        {
            skip: email === '' || !clicked,
            refetchOnMountOrArgChange: true
        }
    );

    const handleTextField = useCallback(
        (e: React.FormEvent<HTMLInputElement>) => {
            const { name, value } = e.target as HTMLInputElement;
            const err = userData.errors;
            switch (name) {
                case 'email':
                    err.email = performValidation('emailId', value);
                    break;
                case 'password':
                    err.password = performValidation('loginPassword', value);
            }
            setUserData({
                ...userData,
                [name]: value,
                errors: err
            });
        },
        [userData]
    );

    const validateData = useCallback(() => {
        let valid = true;
        Object.values(userData.errors).forEach(val => val.length > 0 && (valid = false));
        return valid;
    }, [userData.errors]);

    const logInHandle = (e: React.FormEvent) => {
        e.preventDefault();
        const err = userData.errors;
        err.email = performValidation('emailId', userData.email);
        err.password = performValidation('loginPassword', userData.password);

        setUserData({
            ...userData,
            errors: err
        });
        if (validateData()) {
            getAccessToken({
                path: '/super-admin',
                data: {
                    username: userData.email,
                    password: userData.password
                }
            });
            setEmail(userData.email);
        }
    };

    const passwordVisibleHandler = () => {
        setPasswordVisible(!passwordVisible);
    };

    useEffect(() => {
        if (tokenData && Object.values(tokenData).length > 0) {
            setClicked(true);
            const tokenInfo = {
                ...tokenData.authentication.attributes,
                email: tokenData.authentication.name,
                authenticated: tokenData.authenticated
            };
            localStorage.setItem('token', JSON.stringify(tokenInfo));
        } else if (loginError) {
            const errorObj = loginError as ErrorType;
            addToast({
                variant: 'error',
                header: errorObj.data?.message || 'Something went wrong. Please try again later.',
                timer: 3000
            });
            if (errorObj.data.message === 'Please add organisation details!') {
                navigateTo(routeConstants.companyInfoPage, {
                    state: { email }
                });
            }
        }
    }, [email, loginError, navigateTo, tokenData]);

    const isDisabled = useMemo(
        () => !(userData && userData.password.length > 0 && userData.email.length > 0 && validateData()),
        [userData, validateData]
    );

    useEffect(() => {
        if (state && state.error) {
            addToast({
                variant: 'error',
                header: state.header,
                message: state.message,
                timer: 3000
            });
            window.history.replaceState({}, document.title);
        }
    }, [state]);

    useEffect(() => {
        if (data) {
            dispatch(
                addUser({
                    id: data.id,
                    organisationId: 0,
                    firstName: data.firstName,
                    lastName: data.lastName,
                    email,
                    phoneNumber: data.contactNo,
                    employeeId: '',
                    roleName: '',
                    teamName: '',
                    modulePermission: data.modulePermission || [],
                    designationName: '',
                    departmentName: '',
                    isLoggedIn: true,
                    onboardingFlow: false,
                    isSuperAdmin: true
                })
            );
            navigateTo('/admin');
        }
    }, [data, dispatch, email, navigateTo]);

    useEffect(() => {
        if ((error && (error as any)?.status === 401) || (error as any)?.status === 502) {
            const errorResponse = error as any;
            addToast({
                variant: 'error',
                header: errorResponse.message || 'Unauthorized access! Please contact system administrator/HR.',
                timer: 3000
            });
            setEmail('');
            setClicked(false);
        }
    }, [error, navigateTo]);

    return {
        isUserDataLoading,
        isFetchingToken,
        userDetails,
        logInHandle,
        handleTextField,
        passwordVisibleHandler,
        passwordVisible,
        userData,
        isDisabled
    };
};
