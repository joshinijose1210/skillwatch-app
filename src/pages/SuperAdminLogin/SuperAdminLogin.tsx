import { Logo } from '@common';
import { routeConstants } from '@constants';
import { Text, TextField } from '@medly-components/core';
import { RemoveRedEyeIcon } from '@medly-components/icons';
import { Navigate } from 'react-router-dom';
import { ReactComponent as SkillWatchLogo } from '../../constants/images/logos/Skillwatch.svg';
import Hand from '../../constants/images/logos/hand.png';
import * as Styled from './SuperAdminLogin.styled';
import { useSuperAdminLogin } from './useSuperAdminLogin';
import { useAutoLoginLogout } from '@common/hooks/useAutoLoginLogout';

export const SuperAdminLogin = () => {
    const {
        isFetchingToken,
        isDisabled,
        userDetails,
        logInHandle,
        userData,
        handleTextField,
        isUserDataLoading,
        passwordVisible,
        passwordVisibleHandler
    } = useSuperAdminLogin();
    useAutoLoginLogout();

    if (!userDetails.isLoggedIn) {
        return (
            <Styled.StylePageContent>
                <Styled.LogoWrapper>
                    <Logo href={process.env.LOGIN_URL}>
                        <SkillWatchLogo />
                    </Logo>
                </Styled.LogoWrapper>
                <Styled.RowWrapper>
                    <Styled.Login>
                        <Styled.StyledDiv>
                            <Styled.LoginRight>
                                <Styled.RowWrapper>
                                    <Text textVariant="h2" textWeight="Medium">
                                        Hello there
                                    </Text>
                                    <Styled.StyleHandImage src={Hand} alt="hand" />
                                </Styled.RowWrapper>
                                <Styled.LoginForm onSubmit={logInHandle} autoComplete="off">
                                    <TextField
                                        size="M"
                                        type="name"
                                        variant="outlined"
                                        name="email"
                                        label="Email"
                                        data-testid="emailInput"
                                        fullWidth={true}
                                        value={userData.email}
                                        onChange={e => handleTextField(e)}
                                        errorText={userData.errors.email}
                                    />
                                    <TextField
                                        size="M"
                                        type={passwordVisible ? 'text' : 'password'}
                                        variant="outlined"
                                        name="password"
                                        label="Password"
                                        data-testid="passwordInput"
                                        fullWidth={true}
                                        value={userData.password}
                                        onChange={e => handleTextField(e)}
                                        errorText={userData.errors.password}
                                        suffix={() => (
                                            <RemoveRedEyeIcon
                                                data-testid="passwordVisibleHandler"
                                                onClick={passwordVisibleHandler}
                                                iconColor={passwordVisible ? 'blue' : 'grey'}
                                            />
                                        )}
                                    />
                                    <Styled.StyleButton
                                        type="submit"
                                        data-testid="submitButton"
                                        size="M"
                                        fullWidth={true}
                                        disabled={isDisabled}
                                        isLoading={isFetchingToken || isUserDataLoading}
                                    >
                                        Login
                                    </Styled.StyleButton>
                                </Styled.LoginForm>
                            </Styled.LoginRight>
                        </Styled.StyledDiv>
                    </Styled.Login>
                </Styled.RowWrapper>
            </Styled.StylePageContent>
        );
    } else {
        return <Navigate to={routeConstants.root} />;
    }
};
