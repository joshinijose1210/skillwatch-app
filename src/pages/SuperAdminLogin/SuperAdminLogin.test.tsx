import { RTKRender, cleanup, fireEvent, screen, waitFor } from '@utils/test-utils';
import { SuperAdminLogin } from './SuperAdminLogin';
import { routeConstants } from '@constants';
import MockAdapter from 'axios-mock-adapter';
import { apiInstance } from '@utils';
import * as reactRouterDom from 'react-router-dom';
import { SuperAdminLoginMockData } from '@mocks/superAdminLogin';
const mock = new MockAdapter(apiInstance, { onNoMatch: 'throwException' });

const mockNavigate = jest.fn();

const useLocationMock = jest.spyOn(reactRouterDom, 'useLocation');

jest.mock('react-router-dom', () => ({
    ...(jest.requireActual('react-router-dom') as any),
    useNavigate: () => mockNavigate,
    useLocation: jest.fn(),
    Navigate: jest.fn(() => <div data-testid="mock-navigate">Navigate component</div>)
}));

beforeEach(mock.reset);

afterEach(cleanup);

describe('Super admin login page', () => {
    beforeEach(() => {
        useLocationMock.mockReturnValue({
            pathname: '/',
            state: null,
            search: '',
            hash: '',
            key: ''
        });
    });

    it('should take snapshot', () => {
        const { initialState } = SuperAdminLoginMockData;
        const { container } = RTKRender(<SuperAdminLogin />, { initialState: { ...initialState } });
        expect(container).toMatchSnapshot();
    });

    it('should show error toast when some error occurs', async () => {
        useLocationMock.mockReturnValue({
            pathname: '/',
            state: {
                error: true,
                header: 'thisHeader',
                message: 'thisErrorMessage'
            },
            search: '',
            hash: '',
            key: ''
        });
        RTKRender(<SuperAdminLogin />, {
            initialState: { isLoggedIn: false }
        });
        expect(screen.getByText(/thisErrorMessage/i)).toBeInTheDocument();
    });

    it('should validate input credentials properly', () => {
        const { initialState } = SuperAdminLoginMockData;

        RTKRender(<SuperAdminLogin />, { initialState: { ...initialState } });

        const emailInput = screen.getByTestId<HTMLInputElement>('emailInput');
        const passwordInput = screen.getByTestId<HTMLInputElement>('passwordInput');
        const submitButton = screen.getByTestId('submitButton');
        fireEvent.change(emailInput, { target: { value: 'superadmin@company.com ' } });
        fireEvent.change(passwordInput, { target: { value: 'SuperPass124#' } });
        expect(screen.getByText('Please enter valid email address.')).toBeInTheDocument();
        expect(submitButton).toBeDisabled();
        fireEvent.change(emailInput, { target: { value: 'superadmin@company.com' } });
        expect(submitButton).toBeEnabled();
    });

    it('should login properly', async () => {
        const { initialState, get, post } = SuperAdminLoginMockData;

        mock.onPost(post.url).replyOnce(200, {
            ...post.responseData
        });
        mock.onGet(get.url).replyOnce(200, {
            ...get.responseData
        });
        RTKRender(<SuperAdminLogin />, { initialState: { ...initialState } });

        const emailInput = screen.getByTestId<HTMLInputElement>('emailInput');
        const passwordInput = screen.getByTestId<HTMLInputElement>('passwordInput');
        const submitButton = screen.getByTestId('submitButton');
        fireEvent.change(emailInput, { target: { value: 'superadmin@company.com' } });
        fireEvent.change(passwordInput, { target: { value: 'SuperPass124#' } });
        fireEvent.click(submitButton);
        await waitFor(() => expect(mockNavigate).toHaveBeenCalledWith('/admin'));
    });

    it('should not login when something goes wrong', async () => {
        const { initialState, post } = SuperAdminLoginMockData;

        mock.onPost(post.url).replyOnce(404);
        RTKRender(<SuperAdminLogin />, { initialState: { ...initialState } });
        const emailInput = screen.getByTestId('emailInput') as HTMLInputElement;
        const passwordInput = screen.getByTestId('passwordInput') as HTMLInputElement;
        const submitButton = screen.getByTestId('submitButton');
        fireEvent.change(emailInput, { target: { value: 'superadmin@company.com' } });
        fireEvent.change(passwordInput, { target: { value: 'SuperPass124#' } });
        fireEvent.click(submitButton);
        expect(await screen.findByText('Something went wrong. Please try again later.')).toBeInTheDocument();
    });

    it('should not login when unauthorized and show unauthorized error message', async () => {
        const { initialState, post } = SuperAdminLoginMockData;

        mock.onPost(post.url).replyOnce(200, {
            ...post.responseData
        });
        mock.onGet('https://dummy-api.example.com/api/employees/email?emailId=superadmin@company.com').replyOnce(401);
        RTKRender(<SuperAdminLogin />, { initialState: { ...initialState } });
        const emailInput = screen.getByTestId('emailInput') as HTMLInputElement;
        const passwordInput = screen.getByTestId('passwordInput') as HTMLInputElement;
        const submitButton = screen.getByTestId('submitButton');
        fireEvent.change(emailInput, { target: { value: 'superadmin@company.com' } });
        fireEvent.change(passwordInput, { target: { value: 'SuperPass124#' } });
        fireEvent.click(submitButton);
        expect(await screen.findByText('Unauthorized access! Please contact system administrator/HR.')).toBeInTheDocument();
    });

    it('should toggle the password visibility', async () => {
        const { initialState } = SuperAdminLoginMockData;

        RTKRender(<SuperAdminLogin />, { initialState: { ...initialState } });
        const passwordInput = screen.getByTestId<HTMLInputElement>('passwordInput');
        const passwordVisibleHandler = screen.getByTestId('passwordVisibleHandler');
        expect(passwordVisibleHandler).toBeInTheDocument();
        await waitFor(() => expect(passwordInput.type).toBe('password'));
        fireEvent.click(passwordVisibleHandler);
        await waitFor(() => expect(passwordInput.type).toBe('text'));
    });

    it('should navigate to company info page when no org is added', async () => {
        const { initialState, post } = SuperAdminLoginMockData;

        mock.onPost(post.url).replyOnce(404, {
            message: 'Please add organisation details!'
        });
        RTKRender(<SuperAdminLogin />, { initialState: { ...initialState } });
        const emailInput = screen.getByTestId('emailInput') as HTMLInputElement;
        const passwordInput = screen.getByTestId('passwordInput') as HTMLInputElement;
        const submitButton = screen.getByTestId('submitButton');
        fireEvent.change(emailInput, { target: { value: 'superadmin@company.com' } });
        fireEvent.change(passwordInput, { target: { value: 'SuperPass124#' } });
        fireEvent.click(submitButton);
        expect(await screen.findByText('Please add organisation details!')).toBeInTheDocument();
        expect(mockNavigate).toHaveBeenCalledWith(routeConstants.companyInfoPage, { state: { email: 'superadmin@company.com' } });
    });

    it('should redirect to root page when already logged in', async () => {
        RTKRender(<SuperAdminLogin />, { initialState: { isLoggedIn: true } });
        expect(await screen.findByText('Navigate component')).toBeInTheDocument();
    });
});
