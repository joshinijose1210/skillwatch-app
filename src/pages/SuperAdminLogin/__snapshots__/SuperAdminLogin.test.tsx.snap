// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Super admin login page should take snapshot 1`] = `
<div>
  <main
    class="sc-1ehavxz-0 dDyhBJ sc-4t4v5e-0 cqopww"
  >
    <div
      class="sc-4t4v5e-13 gNhnyh"
    >
      <a
        class="sc-tkiih6-18 fdIgWN"
      >
        <svg
          fill="none"
          height="63"
          viewBox="0 0 321 63"
          width="321"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M28.977 52.084c0 3.42-1.288 6.105-3.863 8.06C22.587 62.047 18.894 63 14.034 63c-3.936 0-7.167-.586-9.694-1.758C1.862 60.07.623 58.507.623 56.553c0-.928.194-1.733.583-2.417a5.533 5.533 0 0 1 1.676-1.685c1.36.879 2.916 1.66 4.665 2.344 1.798.684 3.863 1.026 6.196 1.026 4.033 0 6.05-1.246 6.05-3.737 0-1.025-.365-1.856-1.094-2.49-.68-.635-1.773-1.124-3.28-1.466l-4.082-.952c-3.595-.782-6.268-2.003-8.018-3.663-1.7-1.71-2.55-3.956-2.55-6.74 0-3.175 1.263-5.739 3.79-7.692 2.527-2.003 6.074-3.004 10.642-3.004 2.332 0 4.397.244 6.195.733 1.847.44 3.305 1.099 4.374 1.978 1.069.879 1.603 1.929 1.603 3.15 0 .879-.194 1.66-.583 2.344a4.216 4.216 0 0 1-1.385 1.539c-.583-.342-1.385-.709-2.405-1.1a26.109 26.109 0 0 0-3.426-1.025 16.422 16.422 0 0 0-3.717-.44c-1.896 0-3.378.294-4.447.88-1.069.586-1.603 1.465-1.603 2.637 0 .781.291 1.465.874 2.051.632.538 1.701.977 3.208 1.319l3.571.88c4.034.927 6.95 2.295 8.747 4.102 1.847 1.758 2.77 4.078 2.77 6.96zM49.73 47.396l-6.413-4.323L59.79 26.59c1.895 0 3.377.415 4.446 1.245 1.118.83 1.676 1.88 1.676 3.15 0 1.124-.364 2.1-1.093 2.93-.68.831-1.774 1.881-3.28 3.151l-11.808 10.33zm-6.195-2.93l7.143-3.37 16.911 15.97c-.146 1.71-.656 3.053-1.53 4.03-.827.927-2.017 1.391-3.572 1.391a5.597 5.597 0 0 1-3.062-.879c-.923-.586-1.92-1.514-2.988-2.784L43.535 44.465zm-7.143-2.052l9.257.44v18.974c-.389.098-.972.22-1.75.366-.728.147-1.554.22-2.478.22-1.7 0-2.964-.317-3.79-.952-.826-.635-1.24-1.759-1.24-3.37V42.414zm9.257 5.201l-9.257-.439V14.282c.389-.146.972-.269 1.75-.366.777-.147 1.603-.22 2.477-.22 1.75 0 3.013.317 3.79.952.827.586 1.24 1.71 1.24 3.37v29.597zM74.238 26.022c0-1.465.486-2.686 1.458-3.663 1.02-.977 2.308-1.465 3.863-1.465 1.555 0 2.819.488 3.79 1.465 1.021.977 1.531 2.198 1.531 3.663 0 1.465-.51 2.71-1.53 3.736-.972.977-2.236 1.466-3.79 1.466-1.556 0-2.843-.489-3.864-1.466-.972-1.025-1.458-2.27-1.458-3.736zM155.669 39.41a147.394 147.394 0 0 1-2.333 8.059 391.47 391.47 0 0 1-2.405 7.692c-.729 2.394-1.385 4.372-1.968 5.934-.486.391-1.167.709-2.041.953-.826.293-1.871.44-3.135.44-1.36 0-2.599-.196-3.717-.587-1.069-.44-1.774-1.074-2.114-1.904-.486-.977-1.02-2.37-1.603-4.176a267.37 267.37 0 0 1-1.75-6.227c-.583-2.394-1.19-4.86-1.822-7.4a353.69 353.69 0 0 1-1.604-7.326 279.653 279.653 0 0 1-1.239-6.447c.486-.488 1.142-.928 1.968-1.318.875-.44 1.847-.66 2.916-.66 1.312 0 2.357.318 3.134.953.778.635 1.288 1.71 1.531 3.223.437 2.93.875 5.641 1.312 8.132.486 2.49.923 4.738 1.312 6.74s.729 3.712 1.02 5.128a159.8 159.8 0 0 0 .802 3.224h.292c.291-1.124.777-2.906 1.458-5.348a566.28 566.28 0 0 1 2.405-8.352l2.77-9.524a8.806 8.806 0 0 1 2.114-.879c.777-.244 1.676-.366 2.697-.366 1.312 0 2.405.22 3.28.659.923.44 1.53 1.099 1.822 1.978a301.3 301.3 0 0 1 2.697 9.011l2.187 7.766c.68 2.246 1.142 3.931 1.385 5.055h.364c.826-4.054 1.676-8.328 2.551-12.82a217.616 217.616 0 0 0 2.333-13.48c1.214-.733 2.599-1.1 4.154-1.1 1.264 0 2.284.293 3.062.88.826.537 1.239 1.489 1.239 2.857 0 .634-.146 1.733-.437 3.296-.243 1.514-.608 3.273-1.094 5.275a168.41 168.41 0 0 1-1.53 6.374 603.894 603.894 0 0 1-1.75 6.447 228.604 228.604 0 0 1-1.676 5.64c-.535 1.661-.996 2.955-1.385 3.883-.438.391-1.166.709-2.187.953-1.02.293-2.114.44-3.28.44-3.061 0-4.884-.758-5.467-2.272a461.669 461.669 0 0 0-1.895-5.567 286.27 286.27 0 0 1-2.26-7.327 192.965 192.965 0 0 1-2.113-7.912zM200.996 63c-4.81 0-8.625-.952-11.443-2.857-2.819-1.905-4.228-4.713-4.228-8.425 0-3.272 1.02-5.763 3.061-7.472 2.09-1.759 5.103-2.833 9.039-3.224l9.986-1.026v-1.318c0-1.905-.681-3.272-2.041-4.103-1.361-.879-3.232-1.319-5.613-1.319-1.895 0-3.742.245-5.54.733-1.798.489-3.401 1.05-4.81 1.685-.535-.39-.997-.928-1.385-1.612a4.716 4.716 0 0 1-.51-2.124c0-1.807.971-3.15 2.915-4.03 1.361-.586 2.94-1.025 4.738-1.318a35.29 35.29 0 0 1 5.54-.44c4.762 0 8.552 1.026 11.371 3.077 2.867 2.003 4.3 5.177 4.3 9.524v16.923c0 1.172-.291 2.1-.874 2.784-.535.684-1.264 1.294-2.187 1.832-1.361.781-3.086 1.416-5.175 1.904-2.09.538-4.471.806-7.144.806zm0-7.033c1.41 0 2.697-.146 3.863-.44 1.167-.292 2.017-.61 2.552-.952v-8.132l-7.071.66c-1.943.146-3.426.586-4.446 1.318-1.02.684-1.531 1.71-1.531 3.077 0 1.368.535 2.467 1.604 3.297 1.069.782 2.745 1.172 5.029 1.172zM225.594 42.56h9.111v8.572c0 1.563.462 2.686 1.385 3.37.972.684 2.308 1.026 4.009 1.026.778 0 1.604-.098 2.478-.293a22.635 22.635 0 0 0 2.333-.733c.389.39.705.88.948 1.465.291.538.437 1.172.437 1.905 0 1.465-.656 2.686-1.968 3.663-1.312.928-3.426 1.392-6.342 1.392-3.839 0-6.876-.855-9.111-2.564-2.187-1.758-3.28-4.64-3.28-8.645v-9.157zm5.54-7.399v-7.253h14.286c.243.342.462.83.656 1.466.195.635.292 1.343.292 2.124 0 1.221-.292 2.15-.875 2.784-.583.586-1.36.88-2.332.88h-12.027zm3.571 9.378h-9.111V18.092c.389-.098.948-.22 1.676-.367.778-.146 1.604-.22 2.479-.22 1.701 0 2.94.318 3.717.953.826.635 1.239 1.758 1.239 3.37v22.71zM270.712 33.55c-2.867 0-5.297.952-7.289 2.857-1.944 1.905-2.916 4.64-2.916 8.205 0 3.565.947 6.276 2.843 8.132 1.895 1.856 4.349 2.784 7.362 2.784 1.7 0 3.182-.22 4.446-.66 1.263-.488 2.357-1 3.28-1.538.729.488 1.288 1.05 1.676 1.685.389.586.584 1.319.584 2.198 0 1.66-.997 3.028-2.989 4.102-1.944 1.075-4.544 1.612-7.799 1.612-3.693 0-6.974-.684-9.841-2.051-2.818-1.368-5.005-3.395-6.56-6.08-1.555-2.736-2.332-6.13-2.332-10.184 0-4.103.826-7.522 2.478-10.257 1.701-2.735 3.961-4.786 6.779-6.153 2.818-1.368 5.88-2.052 9.184-2.052 3.256 0 5.856.586 7.799 1.759 1.944 1.123 2.916 2.515 2.916 4.175 0 .782-.194 1.49-.583 2.125a6.242 6.242 0 0 1-1.458 1.538c-.923-.537-2.016-1.025-3.28-1.465-1.215-.488-2.648-.732-4.3-.732zM320.377 47.103h-9.257v-6.667c0-2.442-.656-4.2-1.968-5.275-1.263-1.123-2.891-1.685-4.883-1.685-1.701 0-3.232.318-4.593.953-1.36.635-2.526 1.367-3.498 2.197l-.511-7.472c1.167-.733 2.6-1.416 4.301-2.051 1.749-.635 3.717-.953 5.904-.953 4.373 0 7.872 1.148 10.496 3.444 2.673 2.295 4.009 5.763 4.009 10.402v7.107zm-32.072-4.542h9.258v19.267c-.389.098-.972.22-1.75.366-.729.147-1.555.22-2.478.22-1.701 0-2.964-.317-3.79-.952-.826-.635-1.24-1.759-1.24-3.37V42.56zm22.815 0l9.257.073v19.194c-.388.098-.972.22-1.749.366a12.23 12.23 0 0 1-2.405.22c-1.75 0-3.038-.317-3.864-.952-.826-.635-1.239-1.759-1.239-3.37V42.56zm-13.557 4.761h-9.258v-33.04c.389-.146.972-.269 1.75-.366.777-.147 1.603-.22 2.478-.22 1.749 0 3.013.317 3.79.952.827.586 1.24 1.71 1.24 3.37v29.304z"
            fill="#3872D2"
          />
          <path
            d="M94.143 42.163l9.121.441V61.5c-.386.098-.965.22-1.737.367-.724.147-1.521.22-2.39.22-1.737 0-3.015-.318-3.836-.955-.772-.637-1.158-1.764-1.158-3.382V42.163zm9.121 5.146l-9.121-.367V11.145c.386-.147.941-.27 1.665-.368a13.15 13.15 0 0 1 2.461-.22c1.737 0 2.992.318 3.764.955.821.589 1.231 1.716 1.231 3.383v32.414zM113.442 42.163l9.121.441V61.5c-.386.098-.965.22-1.737.367a12.03 12.03 0 0 1-2.389.22c-1.738 0-3.016-.318-3.837-.955-.772-.637-1.158-1.764-1.158-3.382V42.163zm9.121 5.146l-9.121-.367V.588c.386-.147.941-.27 1.665-.367A13.14 13.14 0 0 1 117.568 0c1.737 0 2.992.319 3.764.956.821.588 1.231 1.715 1.231 3.382v42.971zM75.593 42.163l9.121.441V61.5c-.386.098-.965.22-1.737.367-.724.147-1.52.22-2.389.22-1.737 0-3.016-.318-3.837-.955-.772-.637-1.158-1.764-1.158-3.382V42.163zm9.121 5.146l-9.12-.367V34.898c.385-.147.94-.27 1.664-.368a13.15 13.15 0 0 1 2.461-.22c1.738 0 2.993.318 3.765.955.82.588 1.23 1.716 1.23 3.382v8.662z"
            fill="#97BBF6"
          />
        </svg>
      </a>
    </div>
    <div
      class="sc-4t4v5e-5 jdSQWD"
    >
      <div
        class="sc-4t4v5e-1 cnRZlG"
      >
        <div
          class="sc-4t4v5e-3"
        >
          <div
            class="sc-4t4v5e-4 dczfaz"
          >
            <div
              class="sc-4t4v5e-5 jdSQWD"
            >
              <h2
                class="sc-1ntgl3l-0 hJdvRC"
                textalign="initial"
                textcolor="inherit"
                textvariant="h2"
                textweight="Medium"
              >
                Hello there
              </h2>
              <img
                alt="hand"
                class="sc-4t4v5e-8 eRVhAX"
                src="[object Object]"
              />
            </div>
            <form
              autocomplete="off"
              class="sc-4t4v5e-6 cIMPub"
            >
              <div
                class="sc-17d4s39-0 fGqzZg"
                id="medly-textField-input-wrapper"
                minwidth="20rem"
              >
                <div
                  class="sc-19px6zp-0 BrSGZ"
                  filled="[object Object]"
                  fusion="[object Object]"
                  minrows="1"
                  outlined="[object Object]"
                  textvariant="[object Object]"
                  variant="outlined"
                >
                  <div
                    class="sc-lhr232-0 btNOKw"
                    variant="outlined"
                  >
                    <input
                      aria-describedby="medly-textField-helper-text"
                      class="sc-hb99wz-0 jaEbhp"
                      data-testid="emailInput"
                      errortext=""
                      id="medly-textField-input"
                      inputsize="M"
                      name="email"
                      placeholder=" "
                      type="name"
                      value=""
                      variant="outlined"
                    />
                    <label
                      class="sc-1du51u0-0 eMZrST"
                      for="medly-textField-input"
                      inputwidth="0"
                      variant="outlined"
                    >
                      Email
                    </label>
                  </div>
                </div>
                
              </div>
              <div
                class="sc-17d4s39-0 fGqzZg"
                id="medly-textField-input-wrapper"
                minwidth="20rem"
              >
                <div
                  class="sc-19px6zp-0 BrSGZ"
                  filled="[object Object]"
                  fusion="[object Object]"
                  minrows="1"
                  outlined="[object Object]"
                  textvariant="[object Object]"
                  variant="outlined"
                >
                  <div
                    class="sc-lhr232-0 btNOKw"
                    variant="outlined"
                  >
                    <input
                      aria-describedby="medly-textField-helper-text"
                      class="sc-hb99wz-0 jaEbhp"
                      data-testid="passwordInput"
                      errortext=""
                      id="medly-textField-input"
                      inputsize="M"
                      name="password"
                      placeholder=" "
                      type="password"
                      value=""
                      variant="outlined"
                    />
                    <label
                      class="sc-1du51u0-0 eMZrST"
                      for="medly-textField-input"
                      inputwidth="0"
                      variant="outlined"
                    >
                      Password
                    </label>
                  </div>
                  <svg
                    class="sc-1pf3x8a-0 etlwIk"
                    data-testid="passwordVisibleHandler"
                    fill="none"
                    height="1em"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm-3-5c0-1.66 1.34-3 3-3s3 1.34 3 3-1.34 3-3 3-3-1.34-3-3z"
                      fill="#000"
                      fill-opacity=".54"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
                
              </div>
              <button
                class="sc-1k2druj-0 hGwSqY sc-4t4v5e-12 iyeUzs"
                data-testid="submitButton"
                disabled=""
                edges="square"
                type="submit"
                variant="solid"
              >
                <span
                  class="sc-1ntgl3l-0 eOsHHD"
                  textalign="initial"
                  textcolor="inherit"
                  textvariant="button1"
                >
                  Login
                </span>
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
